Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -471,9 +471,21 @@ StringView ProcessorBase<T>::platform_string()
 }
 
 template<typename T>
-void ProcessorBase<T>::wait_for_interrupt() const
+void ProcessorBase<T>::idle() const
 {
-    asm("wfi");
+    // This function is expected to be called with interrupts disabled.
```

Comment:
Can't we use `VERIFY_INTERRUPTS_DISABLED()`?
Or maybe we need `VERIFY(T::::are_interrupts_enabled())`
(Same for all architectures)